evm_trace/gas.py

```diff
@@ -0,0 +1,54 @@
+from typing import Dict, List
+
+from evm_trace.base import CallTreeNode
+
+Address = str
+Calldata = bytes
+GasUsage = int
+
+GasReport = Dict[Address, Dict[Calldata, List[GasUsage]]]
+
+
+def extract_gas_report(calltree: CallTreeNode) -> GasReport:
+    """
+    Extracts a gas report object from a :class:`~evm_trace.base.CallTreeNode`
+
+    Args:
+        calltree (:class:`~evm_trace.base.CallTreeNode`): extractable call tree
+
+    Returns:
+        :class:`~evm_trace.gas.GasReport`: Gas report structure from a call tree.
+    """
+    report = {
+        calltree.address: {calltree.calldata: [calltree.gas_cost] if calltree.gas_cost else []}
+    }
+
+    for call in calltree.calls:
+        report = combine_gas_reports(report, extract_gas_report(call))
```

Comment:
Just a nit, use `node` here
```suggestion
    for node in calltree.calls:
        report = combine_gas_reports(report, extract_gas_report(node))
```